calendar_public_holiday/models/calendar_public_holiday.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2015 2011,2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
+# Copyright 2020 InitOS Gmbh
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResourceCalendarPublicHoliday(models.Model):
+    _name = "calendar.public.holiday"
+    _description = "Calendar Public Holiday"
+    _rec_name = "year"
+    _order = "year"
+
+    year = fields.Integer(
+        "Calendar Year", required=True, default=fields.Date.today().year
```

Comment:
If not, I think this is going to be a static value.

```suggestion
        "Calendar Year", required=True, default=lambda self: fields.Date.today().year
```